docs/cookbook/languages/go.md

```diff
@@ -12,12 +12,17 @@ See the [builds][build-concept] concept page for more details.
 
 Since the output of the build must be copied to the `$out` directory, you may either install the output directly to `$out`, or you may copy the executable there manually after running `go build`.
 
+Go adds metadata to compiled binaries that allows details from the build environment to leak through.
+For example, a compiled binary will contain absolute paths to source files.
```

Comment:
nonblocking question: are source files the issue or the go compiler itself?

Reply:
This is just an example of _a_ thing the Go compiler embeds.

Reply:
My understanding of the original issue is that by including its own path in the binary it masks the problem of some dependencies being missing from the closure because those dependencies are in the Go closure.